crates/signs/src/bars.rs

```diff
@@ -269,6 +293,23 @@ fn update_visibility_distance(
     }
 }
 
+fn update_visibility_timer(
```

Comment:
I think that the implementation might be a bit simplified and made more efficient with the following changes to this commit:

- Change `spawn()` system so that each bar has `BarUpdateTimer` component.
- Change `update_visibility_timer` so that 
  - it iterates over all `BarUpdateTimer` components and changes visibility flag `TIMER_BAR_ID` to `false` if the timer [just_finished](https://docs.rs/bevy/latest/bevy/prelude/struct.Timer.html#method.just_finished) (among other things to avoid unnecessary change detection).
  - before calling `tick()` it (conditionally to avoid change detection) changes `TIMER_BAR_ID` to `true` if `elapsed()` time of the timer is 0
- Change `update_value()` system so that it does not insert `BarUpdateTimer` component (see first point, we can avoid frequent re-adding by keeping the component forever). It should only reset the timer.

Reply:
I figured that might be the case, just decided to do the first approach that came to mind :)